clustermesh/clustermesh.go

```diff
@@ -1811,7 +1811,27 @@ func DisableWithHelm(ctx context.Context, k8sClient *k8s.Client, params Paramete
 	return err
 }
 
+// ConnectWithHelm enables clustermesh using a Helm Upgrade action Certificates are generated via
+// the Helm chart's cronJob (certgen) mode As with classic mode, only autodetected IP-based
+// clustermesh-apiserver Service endpoints are currently supported.
 func (k *K8sClusterMesh) ConnectWithHelm(ctx context.Context) error {
+	v, err := k.client.GetRunningCiliumVersion(ctx, k.params.Namespace)
+	if err != nil {
+		return err
+	}
+	ciliumVer, err := utils.ParseCiliumVersion(v)
+	if err != nil {
+		k.Log("⚠️ Failed to parse Cilium version %s: %w. "+
+			"1.14+ is assumed. Continuing with Helm-mode", v, err)
+	} else {
+		if ciliumVer.LT(versioncheck.MustVersion("1.14.0")) {
```

Comment:
Snapshot versions seem not to be included by this check (I'm not sure if that is intentional).

Reply:
The tricky thing here is treating snapshots of < 1.14 and 1.14 appropriately. We have a function that attempts to exec into a pod to grab the version, which I was using here previously, but it seemed like overkill.

I’ll think about this more.

Reply:
try `>=1.14.0-dev`.

it should work with `1.14.0-snapshot.*` versions as well as the main branch chart https://github.com/cilium/cilium/blob/main/VERSION